test/0.8.9/oracle/validators-exit-bus-oracle-access-control.test.js

```diff
@@ -0,0 +1,169 @@
+const PUBKEYS = [
+  '0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+  '0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb',
+  '0xcccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc',
+  '0xdddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd',
+  '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+]
+
+contract('ValidatorsExitBusOracle', ([admin, member1, member2, member3, account1, stranger]) => {
+  let consensus
+  let oracle
+  let oracleVersion
+  let initTx
+  let exitRequests
+  let reportFields
+  let reportItems
+  let reportHash
+
+  const submitDataRoleKeccak156 = web3.utils.keccak256('SUBMIT_DATA_ROLE')
+  const pauseRoleKeccak156 = web3.utils.keccak256('PAUSE_ROLE')
+  const resumeRoleKeccak156 = web3.utils.keccak256('RESUME_ROLE')
+
+  const getReportFields = (override = {}) => ({
+    consensusVersion: CONSENSUS_VERSION,
+    dataFormat: DATA_FORMAT_LIST,
+    ...override
+  })
+  const deploy = async () => {
+    const deployed = await deployExitBusOracle(admin, { resumeAfterDeploy: true })
+    consensus = deployed.consensus
+    oracle = deployed.oracle
+    initTx = deployed.initTx
+
+    oracleVersion = +(await oracle.getContractVersion())
+
+    await consensus.addMember(member1, 1, { from: admin })
+    await consensus.addMember(member2, 2, { from: admin })
+    await consensus.addMember(member3, 2, { from: admin })
+
+    const { refSlot } = await deployed.consensus.getCurrentFrame()
+    exitRequests = [
+      { moduleId: 1, nodeOpId: 0, valIndex: 0, valPubkey: PUBKEYS[0] },
+      { moduleId: 1, nodeOpId: 0, valIndex: 2, valPubkey: PUBKEYS[1] },
+      { moduleId: 2, nodeOpId: 0, valIndex: 1, valPubkey: PUBKEYS[2] }
+    ]
+
+    reportFields = getReportFields({
+      refSlot: +refSlot,
+      requestsCount: exitRequests.length,
+      data: encodeExitRequestsDataList(exitRequests)
+    })
+
+    reportItems = getReportDataItems(reportFields)
+    reportHash = calcReportDataHash(reportItems)
+
+    await deployed.consensus.submitReport(refSlot, reportHash, CONSENSUS_VERSION, { from: member1 })
+    await deployed.consensus.submitReport(refSlot, reportHash, CONSENSUS_VERSION, { from: member3 })
+  }
+
+  context('Access control', () => {
+    context('deploying', () => {
+      before(deploy)
+
+      it('deploying accounting oracle', async () => {
+        assert.isDefined(oracle)
+        assert.isDefined(consensus)
+        assert.isDefined(oracleVersion)
+        assert.isDefined(initTx)
+        assert.isDefined(exitRequests)
+        assert.isDefined(reportFields)
+        assert.isDefined(reportItems)
+        assert.isDefined(reportHash)
+      })
+    })
+    context('DEFAULT_ADMIN_ROLE', () => {
+      beforeEach(deploy)
```

Comment:
I suggest using EVM snapshot here and there to speed-up the tests a little
```js
before() {
    // do all the stuff
    snapshot = new EvmSnapshot(hre.ethers.provider)
    await snapshot.make()
}

afterEach(async () => {
    await snapshot.rollback()
})
```

Reply:
This seems very handy. Can you point to docs/package this is from? I got ambiguous results in google.

Reply:
The motivational example https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/test/0.8.9/staking-router-deposits.test.js#L31,  and the actual `EvmSnapshot` implementation: https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/test/helpers/blockchain.js

`evm_snapshot` and `evm_revert` low-level methods both are from [ganache](https://docs.nethereum.com/en/latest/ethereum-and-clients/ganache-cli/), it's a common practice to use them for test fixtures, especially in brownie and ape.